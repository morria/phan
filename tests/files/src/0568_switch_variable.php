<?php

/**
 * @param ?string $x
 * @param ?int $y
 * @param bool $z
 */
function test_switch($x, $y, $z) {
    switch ($x) {
    case 'a':
        echo intdiv($x, 2);
        break;
    case 'b':
        echo intdiv($x, 2);
        break;
    }
    switch ($y) {
    case 1:
        if (rand() % 2 > 0) {
            echo "odd\n";
            break;
        }
    case 2:
        echo strlen($y);
        echo "Something";
    case 3:
        echo strlen($y);
        break;
    }
    switch (true) {
        case rand() % 2 > 0:
            echo "odd switch\n";
            break;
        case $z:
            echo strlen($z);  // Phan doesn't support this yet, but shouldn't misbehave.
            break;
        default:
            echo "default\n";
            break;
    }
}
