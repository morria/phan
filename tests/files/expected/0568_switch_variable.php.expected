%s:11 PhanTypeMismatchArgumentInternal Argument 1 (numerator) is 'a' but \intdiv() takes int
%s:14 PhanTypeMismatchArgumentInternal Argument 1 (numerator) is 'b' but \intdiv() takes int
%s:24 PhanTypeMismatchArgumentInternal Argument 1 (string) is 1|2 but \strlen() takes string
%s:27 PhanTypeMismatchArgumentInternal Argument 1 (string) is 1|2|3 but \strlen() takes string
%s:35 PhanTypeMismatchArgumentInternal Argument 1 (string) is bool but \strlen() takes string
